webapp/components/Editor/defaultExtensions.spec.js

```diff
@@ -0,0 +1,93 @@
+import defaultExtensions from './defaultExtensions.js'
+import { Editor } from 'tiptap'
+
+let content
+let createEditor
+
+describe('defaultExtensions', () => {
+  describe('editor', () => {
+    createEditor = () => {
+      const componentStub = {
+        placeholder: 'placeholder',
+        $t: jest.fn(),
+        $apollo: {
+          mutate: jest.fn(),
+        },
+      }
+      return new Editor({
+        content,
+        extensions: [...defaultExtensions(componentStub)],
+      })
+    }
+  })
+
+  it('renders', () => {
+    content = ''
+    expect(createEditor().getHTML()).toEqual('<p></p>')
+  })
+
+  describe('`content` contains a mentioning', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content mentioning <a class="mention" data-mention-id="alicias-id" href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" target="_blank">@alicia-luettgen</a>.</p>'
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-01T06:03:32Z" title="Thursday, August 1st 2019, 8:03:32 am +02:00">Aug 1, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,88 @@
+import defaultExtensions from './defaultExtensions.js'
+import { Editor } from 'tiptap'
+
+let content
+let createEditor
+
+describe('defaultExtensions', () => {
+  describe('editor', () => {
+    createEditor = () => {
+      const componentStub = {
+        placeholder: 'placeholder',
+        $t: jest.fn(),
+        $apollo: {
+          mutate: jest.fn(),
+        },
+      }
+      return new Editor({
+        content,
+        extensions: [...defaultExtensions(componentStub)],
+      })
+    }
+  })
+
+  it('renders', () => {
+    content = ''
+    expect(createEditor().getHTML()).toEqual('<p></p>')
+  })
+
+  describe('`content` contains a mentioning', () => {
+    beforeEach(() => {
+      content =
+        '<p>This is a post content mentioning <a class="mention" href="/profile/f0628376-e692-4167-bdb4-d521de5a014f" target="_blank">@alicia-luettgen</a>.</p>'
```


`data-mention-id`?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-01T06:35:07Z" title="Thursday, August 1st 2019, 8:35:07 am +02:00">Aug 1, 2019</time>_

---


is it added in `toDom`?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:14:19Z" title="Thursday, August 1st 2019, 12:14:19 pm +02:00">Aug 1, 2019</time>_

---


Since `parseDOM` is returning empty array, `Mention` and `Hashtag` never get parsed from html. It's a one-way, let's say "write-only".

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:16:49Z" title="Thursday, August 1st 2019, 12:16:49 pm +02:00">Aug 1, 2019</time>_

---


By the way: I was worried I could break the hashtag implementation if hashtags are not parsed:
1. Tag a post with 3 hashtags
2. Edit the post without touching the hashtags
3. Are the hashtags gone now? - No, because the backend implementation searches for `a[href]` with a `href` starting with `/search/hashtag/`.

It's sometimes really, really confusing because all the implementations are interconnected. We have to be very careful about refactoring it.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:16:49Z" title="Thursday, August 1st 2019, 12:16:49 pm +02:00">Aug 1, 2019</time>_

---


By the way: I was worried I could break the hashtag implementation if hashtags are not parsed:
1. Tag a post with 3 hashtags
2. Edit the post without touching the hashtags
3. Are the hashtags gone now? - No, because the backend implementation searches for `a[href]` with a `href` starting with `/search/hashtag/`.

It's sometimes really, really confusing because all the implementations are interconnected. We have to be very careful about refactoring it.